src/app/authorization/registration/registration.component.ts

```diff
@@ -63,53 +56,33 @@ export class RegistrationComponent {
     });
     localStorage.setItem('registeruser', JSON.stringify(this.registeruser));
     this.route.navigateByUrl('login');
-  }
-
 
-  initForm() {
-    this.registrationForm = new FormGroup(
-      {
-        id: new FormControl(Date.now()),
-        fullname: new FormControl('', [
-          Validators.required,
-          // Validators.pattern('[a-zA-Z-]'),
-        ]),
-
-        pwd: new FormControl('', [
-          Validators.required,
-          Validators.minLength(6),
-          Validators.maxLength(15),
-        ]),
-        cpwd: new FormControl('', [
-          Validators.required,
-          Validators.minLength(6),
-          Validators.maxLength(15),
-        ]),
-        gender: new FormControl('', [Validators.required]),
-        birthday: new FormControl('', [Validators.required]),
-        mobile: new FormControl('', [
-          Validators.required,
-          Validators.pattern('[0-9]*'),
-          Validators.minLength(10),
-          Validators.maxLength(10),
-        ]),
-        email: new FormControl('', [Validators.required, Validators.email]),
-      },
-      { validators: this.identityRevealedValidator }
-    );
+    this.submitted = true;
+    if (this.registrationForm.invalid) {
+      return;
+    }
   }
+
   identityRevealedValidator: ValidatorFn = (
-    control: AbstractControl
+    controls: AbstractControl
   ): ValidationErrors | null => {
-    const pwd = control.get('pwd');
-    const cpwd = control.get('cpwd');
-    return pwd && cpwd && pwd.value !== cpwd.value
-      ? { identityRevealed: true }
-      : null;
+    const pwd = this.registrationForm?.controls['pwd'];
+    return pwd?.value !== controls?.value ? { identityRevealed: true } : null;
   };
+
+  validateNumber: ValidatorFn = (
+    controls: AbstractControl
+  ): ValidationErrors | null => {
+    const regex = /^(?:(?:\+|0{0,2})91(\s*[\-]\s*)?|[0]?)?[123456789]\d{9}$/;
+    return regex.test(controls.value) ? null : { pattern: true };
+  };
+
   getToday(): string {
     return new Date().toISOString().split('T')[0];
```

Comment:
Could you please change this to new Date().toISOString().slice(0, 10) as split can trow error if date is null

Reply:
YES, I have changed